### ISSUE TITLE
Many standard library types not found

### ISSUE THREAD

**Issue by jminer** [rust-analyzer/rust-analyzer#2154]
I just updated to the latest nightly compiler and rust-analyzer (eb8f76a86fe88fcad370577b344f397dd6118cfd). I'm using `i686-pc-windows-msvc` on Windows 10: `rustc 1.40.0-nightly (87cbf0a54 2019-11-01)`

For me currently, rust-analyzer can't find many items in `std` like `vec![]`, `Vec`, or `Option`. On the other hand, it can find `Range` and `HashMap`. I thought these are supposed to work and did before I updated. If I run analysis-stats on rust-analyzer:

```
Total expressions: 78546
Expressions of unknown type: 21367 (27%)
Expressions of partially unknown type: 3931 (5%)
```

which is a lot worse than other people recently posted. Could this be related to any recent nightly changes or that I'm on Windows?

I'm not using rustup (installing msi and unpacking the src archive manually), but I think the src is in the right place since ra does find some types in `std`.

**Comment by edwin0cheng** [comment]
I did try it in my Windows machine (eb8f76a) with rustc 1.40.0-nightly (87cbf0a54 2019-11-01) : 

```
cargo +nightly run --release -p ra_cli -- analysis-stats .
```

Here is the result :

```
Total expressions: 89603
Expressions of unknown type: 5678 (6%)
Expressions of partially unknown type: 3407 (3%)
Type mismatches: 3724
Inference: 29.0840339s, 0b allocated 0b resident
Total: 40.423245s, 0b allocated 0b resident
```

**Comment by flodiebold** [comment]
:thinking: 
One thing to check would be to use 'go to definition' on an std type that does resolve (e.g.) `HashMap` and see where it takes you -- does it go to the correct installation of rust?
Apart from that, you could try creating a very small test project that references just e.g. `Vec`, and run `ra_cli` with `RUST_LOG=ra_hir::nameres=debug`. That will probably still produce a lot of logs, but might give a hint as to what's happening.

**Comment by jminer** [comment]
@edwin0cheng Thanks for trying it. It was useful to know it had nothing to do with Windows or the version of Rust.

@flodiebold Yes, `HashMap` and other types took me to the definition in the right file in the right version of Rust. I created a project and ran `ra_cli` as you suggested, and it dumped 27k lines of output. From searching for `Vec` in it, it seems like nameres can resolve `vec::Vec` in some of the standard library files, but not in the test file.

It seems it is because I'm not using Rustup. I tried Rustup in a VM and found that the source archive it downloads is different than the source archive on https://forge.rust-lang.org/infra/other-installation-methods.html.

```
Download page:   https://static.rust-lang.org/dist/rust-nightly-src.tar.gz
Rustup manifest: https://static.rust-lang.org/dist/2019-11-04/rust-src-nightly.tar.gz
```

The one on the download page seems to contain many more files and is 20MB, but the one Rustup downloads is only 3.6MB. Looking at it more, it seems like the one on the download page is from Jan 2, 2015! That's what the modified dates in the archive are, and the files have "Copyright 2014" headers. The link must have been changed to point at 2015 source in the past couple months because I still have the previous file (rustc-nightly-src.tar.gz) and it's definitely from 2019. rust-analyzer works much better with the 2019 source archive. Thanks for the help and all the work on rust-analyzer!
